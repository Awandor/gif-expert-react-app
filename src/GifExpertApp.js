// ========================================
// Functional Component
// ========================================

import { useState } from "react";
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ( { defaultCategories = [] } ) => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    // const [categories, setCategories] = useState( ['One Punch', 'Samurai X', 'Dragon Ball'] );
    
    // const [categories, setCategories] = useState( ['One Punch'] );
    
    const [categories, setCategories] = useState( defaultCategories );

    console.log( 'categories: ', categories );

    /* const handleAdd = () => {

        // Para añadir elementos a array hay dos formas

        // setCategories( [...categories, 'Wolverine'] );

        setCategories( estadoAnterior =>{

            return [...estadoAnterior, 'Wolverine'];

        } );

    } */

    return (
        <>
            <h2>Gif Expert App</h2>
            <hr />

            {/* <button onClick={handleAdd}>Agregar</button> */}

            <AddCategory setCategories={setCategories} />

            <ol>
                {
                    categories.map( ( elem ) => {

                        // key no debe ser el index, debe de ser único, normalmente de base de datos

                        // return <li key={elem}>{elem}</li>

                        return <GifGrid key={elem} category={elem} />

                    } )
                }
            </ol>
        </>
    );

};

export default GifExpertApp;