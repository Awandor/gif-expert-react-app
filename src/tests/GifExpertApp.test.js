import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe( 'Pruebas en <GifExpertApp />', () => {

    // Este test se hace con Enzyme

    test( 'debe mostrar <GifExpertApp /> correctamente', () => {

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    } );

    test( 'debe mostrar una lista de categorÃ­as', () => {

        const categories = ['Snoopy', 'Dragon Ball'];

        const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> );

        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );

    } );

} );