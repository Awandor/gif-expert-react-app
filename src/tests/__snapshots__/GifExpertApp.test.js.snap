// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Snoopy"
      key="Snoopy"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
